wandb/sdk/launch/create_job.py

```diff
@@ -335,10 +335,19 @@ def _create_repo_metadata(
         entrypoint = rel_entrypoint
 
     # check if requirements.txt exists
-    if not os.path.exists(os.path.join(local_dir, "requirements.txt")):
-        repo_formd = path.replace(entrypoint, "")
+    # start at the location of the python file and recurse up to the git root
```

Comment:
Probably out of scope for this PR but this basic parent recurse would be a nice candidate for a small util function, we could also use it for a dockerfile or python version file for example